### PR TITLE
Dev/nes handling

### PR DESCRIPTION
**Pull request by HaukurPall**
- Installable project using pyproject.toml. We required Python 3.7 (dataclasses).
- Refactored NER tagging code. The models do not work due to changes from "develop".
- Added some tests in `./tests` but will move them to corresponding subdirectories later.
- All but 1 test passes, which attempts to add PoS tags.
- Some files were edited due to `black` configuration being picked up.

### CODE REVIEW

File: src/greynirseq/nicenlp/utils/logits_filter.py [hunk 1]
```diff
@@ -73,7 +65,7 @@ def max_tensor_by_bins(tensor, bins, softmax_by_bin=False):
             if bin_end - bin_start > 0:
                 bin_tensors = F.softmax(bin_tensors)
             else:
-                bin_tensors = F.sigmoid(bin_tensors)
+                bin_tensors = torch.sigmoid(bin_tensors)
```
Comment (vesteinn) [inline src/greynirseq/nicenlp/utils/logits_filter.py:68]: Any reason for this? I understand they are more or less the same but we'd rather not change something as sensitive unless there is a good reason

File: src/greynirseq/ner/aligner.py [hunk 2]
```diff
@@ -13,123 +16,209 @@
...
 
-class ParallelNER:
-    lang_1 = None
-    lang_2 = None
+@dataclass(frozen=True)
+class NERAlignment:
+    """Hold NER alignment results."""
 
-    print_data_file = None
+    cost: float
```
[the remark this answers is not in the document]
Comment (HaukurPall) [reply src/greynirseq/ner/aligner.py:48]: I will change this to distance.

File: poetry.lock [hunk 3]
```diff
@@ -0,0 +1,1654 @@
+[[package]]
```
[the remark this answers is not in the document]
Comment (HaukurPall) [reply poetry.lock:1]: They suggest committing the poetry.lock file: https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control
They do mention that this is not necessary for libraries. 

Let's leave it out for now.

File: src/greynirseq/ner/postagger.py [hunk 4]
```diff
@@ -1,47 +1,51 @@
...
+def add_marker(ner_marker: NERMarkerIdx, tokens: List[str], idx: int, tag: str):
+    """Add a complex marker around NEs.
+
+    HAS SIDE-EFFECTS!
+    """
+    tokens[ner_marker.start_idx] = f"<e:{idx}:{tag}:>{tokens[ner_marker.start_idx]}"
+    if ner_marker.end_idx - ner_marker.start_idx != 1:
+        tokens[ner_marker.end_idx - 1] = f"{tokens[ner_marker.end_idx - 1]}</e{idx}>"
+    else:
+        tokens[ner_marker.start_idx] += f"</e{idx}>"
```
[the remark this answers is not in the document]
Comment (HaukurPall) [reply src/greynirseq/ner/postagger.py:22]: The original code had the same implementation and I thought it would be fine to leave it like this. It saves some memory churning.

File: src/greynirseq/nicenlp/utils/logits_filter.py (hunk 1 above)
Comment (HaukurPall) [reply src/greynirseq/nicenlp/utils/logits_filter.py:68]: There was a deprecation warning, so I changed it.
The current implementation in PyTorch for `F.sigmoid` is: `input.sigmoid()`. I would expect `torch.sigmoid` to behave in the same way.
